regression/simulator/regression_simulator.cpp

```diff
@@ -51,7 +54,7 @@ int main(int /*argc*/, const char* /*argv*/[])
     // Add custom Struct Property
     instance.addProperty(StructProperty("TestStruct", Struct("TestName", dict, TypeManager())));
     // Add custom Argument Info Property
-    auto argInfo = ArgumentInfo_Create(String("TestArgInfo"), CoreType::ctInt);
+    auto argInfo = ArgumentInfo(String("TestArgInfo"), CoreType::ctInt);
```

Comment:
We switched away from them here...
The reason was actually because Device Domain factory changed, while this code should work with both the new and the old implementation. However your other comment suggested leaving the factory as is and adding a new one or moving to a builder pattern. So if we do that, these changes are unwarranted. However I think the code looks a bit cleaner this way (without `_Create`), so I'll leave these modifications in.